crates/uv-resolver/src/pubgrub/dependencies.rs

```diff
@@ -30,14 +30,28 @@ impl PubGrubDependency {
         dev: Option<&'a GroupName>,
         source_name: Option<&'a PackageName>,
     ) -> impl Iterator<Item = Self> + 'a {
-        let iter = if requirement.extras.is_empty() {
-            Either::Left(iter::once(None))
+        let iter = if !requirement.extras.is_empty() {
+            Either::Left(Either::Left(
+                requirement
+                    .extras
+                    .clone()
+                    .into_iter()
+                    .map(|extra| (Some(extra), None)),
+            ))
+        } else if !requirement.groups.is_empty() {
+            Either::Left(Either::Right(
+                requirement
+                    .groups
+                    .clone()
+                    .into_iter()
+                    .map(|group| (None, Some(group))),
+            ))
         } else {
-            Either::Right(requirement.extras.clone().into_iter().map(Some))
+            Either::Right(iter::once((None, None)))
```

Comment:
As with #9582, I suspect you'll want to do the same thing for groups as I did for extras: conditionally add the base package dependency if the group is declared as conflicting.

Reply:
Done.